### ISSUE TITLE
Problem with neighborhoods plugin and loading graph

### ISSUE THREAD

**Issue by J0dla** [jacomyal/sigma.js#715]
Hello,
I  have problem with loading my graph exported to json from gephi into neighborhoods plugin. Arctic.json example works fine but when i change file at db.load line i get blank canvas without an errors in console. 
I can't figure out what is wrong with my file, when i use parser it works fine.
[xtest2.zip](https://github.com/jacomyal/sigma.js/files/142983/xtest2.zip)
I have another problem ... how to iterate on edges, nodes using event listeners ? In old version of sigma there was code that allowed to hover all neighbours of node when mouse point(iterEdges,e.content) , but now these methods were removed :(
`var g = {
nodes: [],
edges: []
};
var s = new sigma({
  graph: g,
  renderer: {
    container: document.getElementById('graph-container'),
    type: 'canvas'
  },
  settings: {
    doubleClickEnabled: false,
    enableEdgeHovering: true,
    edgeHoverColor: 'edge',
    defaultEdgeHoverColor: '#000',
    edgeHoverSizeRatio: 1,
    edgeHoverExtremities: true,
  }
});
sigma.parsers.json('data/xdtest2.json', 
  s,
  function() {
    s.refresh();
  }
);`
I tried s.g.edges.length s.graph.edges.length but i dont know where nodes and edges are stored. 
Last problem i encountered is when i tried use sigma.plugins.relativeSize(s,1);  after parser, the graph had not changed ... 
Thank you.

**Comment by conker84** [comment]
Hi for the neighbours problem use this function:

```javascript
sigma.classes.graph.addMethod('neighbors', function(nodeId) {
    var k,
        neighbors = {},
        index = this.allNeighborsIndex[nodeId] || {};
    
    for (k in index) {
        neighbors[k] = this.nodesIndex[k];
    }
    return neighbors;
});
```

**Comment by oculardexterity** [comment]
Hi,
I am implementing the above code (adding neighbors function to the graph). Is there an obvious reason why `this.allNeighborsIndex[nodeId]` will not return anything? I've tried with numerous nodes (to make sure they do have edges), but none return anything.

I can verify that the node _is_ found by `this.allNeighborsIndex` by calling `this.allNeighborsIndex.has(nodeId)` — but still it apparently has no neighbors!

I am adding nodes one by one — is that an issue? Do I need to re-add this function after building the graph?

**Comment by ramyasrivatsa** [comment]
Hi,
Did you get a solution for the same. I am unable to get this.allNeighborsIndex. ??
index = this.allNeighborsIndex[nodeId] || {} -- this is always 0.

**Comment by oculardexterity** [comment]
Yes, I rebuilt the library — I think this function is broken in the pre-built version.

**Comment by RiseEarly** [comment]
Can you let me know what you did ?? 
Not sure how to get the values of these private attributes : allNeighborsIndex, outNeighboursIndex, etc.

Or did you go with the traditional approach of getting the source and target to identify the neighbors.

**Comment by oculardexterity** [comment]
I can't really remember precisely — a lot of trial and error. As I said, I think the allNeighborsIndex function was broken in the version I was originally using. So I downloaded the full git repository and ran the Build scripts as described in the README. (First I established it was broken by modifying some of the examples, which import all the packages separately.)

**Comment by RiseEarly** [comment]
OK thanks for your quick response. Did you try the neighborhood plugins by linkurious. I was unable to use that as well.

**Comment by danielritter** [comment]
I use also linkurious and I have the same problem with the neighborhood. It does not work.

**Comment by dijana-sagit** [comment]
Hello Everyone, this worked for me:
```javascript
sigma.classes.graph.addMethod('neighbors', function(nodeId) {
		var i,
		    neighbors = {},
		    index = this.allNeighborsIndex.get(nodeId).keyList() || {};
			for (i = 0; i < index.length; i++) {
				neighbors[index[i]] = this.nodesIndex.get(index[i]);
			}
		return neighbors;
});
```

**Comment by danielritter** [comment]
Thanks, this works!
